### PR TITLE
Michaelf/enable disable mgmt 1317

### PR DESCRIPTION
**Pull request by filanov**

### CODE REVIEW

File: internal/host/statemachine.go [hunk 1]
```diff
@@ -90,5 +91,32 @@ func NewHostStateMachine(th *transitionHandler) stateswitch.StateMachine {
...
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeEnableHost,
+		SourceStates: []stateswitch.State{
+			HostStatusDisabled,
+		},
+		DestinationState: HostStatusDiscovering,
+		PostTransition:   th.PostEnableHost,
+	})
+
+	// Enable disconnected host does not affect the state
```
Comment (ronniel1) [inline internal/host/statemachine.go:104]: If it does not affect the state, Maybe it should not be a transission?

Comment (filanov) [reply internal/host/statemachine.go:104]: How do we combine a state when sometimes it have a transition and sometimes we just allow it to run?
We can return an error in this state if it makes more sense?
I did the same trick with install https://github.com/filanov/bm-inventory/blob/master/internal/host/statemachine.go#L47

Comment (filanov) [reply internal/host/statemachine.go:104]: I'm open to suggestions.

Comment (avishayt) [reply internal/host/statemachine.go:104]: Seems a bit to transition from a state to itself when it doesn't show up that way in the state machine diagram.  It make more sense to return 409.

Comment (filanov) [reply internal/host/statemachine.go:104]: Will do for this case.
What about install? there was a requirement to ignore disabled hosts. should it be a transition as i mentioned in the link?

Comment (avishayt) [reply internal/host/statemachine.go]: Yes disabled->disabled makes sense for the install transition

**Pull request merged**